### DISCUSSION TITLE
0.7.0

### DISCUSSION THREAD

**Discussion by kubilus1** [kubilus1/autoortho#414] [Announcements]
## What's Changed
* Show logs in config UI.  Adjust how we kick off execution by @kubilus1 in https://github.com/kubilus1/autoortho/pull/398
* Fix 'duplicate' best chunk tile matches by @kubilus1 in https://github.com/kubilus1/autoortho/pull/409
* Only validate existing sceneries when necessary by @kubilus1 in https://github.com/kubilus1/autoortho/pull/411

**Full Changelog**: https://github.com/kubilus1/autoortho/compare/0.6.1...0.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/kubilus1/autoortho/releases/tag/0.7.0'>0.7.0</a>.</em>

**Comment by rayw45** [comment]
Where do I download this latest version 0.7 for Windows 10 ??  I don't see it in the downloads list  only up to 6.0.1

**Reply by Jeff2122** [reply]
Click the blue 0.7.0 above.

**Reply by rayw45** [reply]
I could not immediately see a clickable 0.7.0 link to download page but now have found it thanks.  I assume I do not need to download the big scenery files again as I had when updating from 4.4. to 6.1 ( I missed  5)

**Reply by rayw45** [reply]
Okay now updated to AO 0.7.0 but as expected it does not fix the need for debug = True sadly.  As the bug has not been identified yet I hardly expected it to be fixed with this version .  There is simply nothing more I can think of to try with Windows 10 that could fix this problem that was not present in 4.4 and that I hear there are several people reporting. 

I will set debug = True and test this latest AO version anyway and just put up with the resulting slow XP12 flight loading time. Anyway I appreciate the sterling work being done by the developers of this game changing XP12 massive improvement. Certainly once loaded and running 6.1 was faster in use than 4.4 so worth the loading time debug delay for sure.

**Reply by HeliMan** [reply]
Instead debug = true you can try change in .autoortho file threading = false. This helped in my case and no issue.

**Reply by rayw45** [reply]
Thanks for the tip HeliMan but sadly after I made debug = False and threading = False I still get the unable to run error when I get to the Reading New Scenery part of loading a flight.  So this made no difference to the problem here so I'm back to debug = True until this is fixed.  Seem to have run out of any other ideas.

**Reply by rayw45** [reply]
BTYW I am using Dokan not WinFSP so not sure which you are using HeliMan,   However, when I make debug and threading = False then I wait 30 minutes reading new scenery but nothing happens even at an airport I have flown a hundred times so has cache tiles stored.  When I try to close XP I can get to the desktop but cannot close XP and have to reboot Windows. When I change to debug and threaded = True then I wait the usual 4 or 5 minutes and all is working as before.   However AO 0.7.0 is otherwise working just fine and does seem faster once up and flying and not seen any bad quality tiles problems here.

**Comment by wakevortex** [comment]
updated from 0.6.0-1. to 0.7 it loads then CTD after about 5 seconds. reverting to previous build all good...

**Reply by kubilus1** [reply]
OS?  Errors in logs?  Anything relevant info wise?

**Comment by wakevortex** [comment]
Hi Sadly I dont have a log, but I can reinstall and get one..
Windows 11 using WinFSP for AO (dokan doesnt work for me for some reason on 0.6.0-1) ...nothing else to add except that all works beautifully with 0.6.0-1 , and then uninstalling that and going to 0.7, the program loads and dissapears after 5 secs or so ... repeatable on every start...
Anything else I can try that would help?

**Reply by kubilus1** [reply]
Did you try refreshing your config file?  Delete C:\Users\<youruser\.autoortho and restart, resetup.  You won't need to reinstall scenery, but you'll need to reset paths.

**Comment by wakevortex** [comment]
OK tried complete app uninstall/reinstall..same issue... when Gui closes, the Dos box also closes with it.. reinstalling 601 its perfect again
Log now attached.. 
[autoortho.log](https://github.com/kubilus1/autoortho/files/12909347/autoortho.log)

**Reply by kubilus1** [reply]
Thanks for the log.  Wondering if you have 'showconfig=False' or something.  The CLI window will no longer be shown, so that's expected.

**Reply by wakevortex** [reply]
showconfig=true. so its not that 
 I cant even see the GUI long enough to change any paths etc... 
"Did you try refreshing your config file? "
Will try that next and report back  !...appreciate your help on this :-)

**Reply by kubilus1** [reply]
Can you set debug=True in the config and re-run and send me the logs.  Thanks!

**Reply by kubilus1** [reply]
Also are you running this from the source code, zip file, or install?

**Reply by wakevortex** [reply]
Will try the debug set to true and let you know...Im using the installer, and uninstalling the previous version first using the uninstaller in the directory...

**Reply by wakevortex** [reply]
heres the log ... 
[autoortho.log](https://github.com/kubilus1/autoortho/files/12910418/autoortho.log)

**Reply by wakevortex** [reply]
will try refreshing config file tomorrow...

**Reply by kubilus1** [reply]
Nothing is jumping out at me in the logs.  Someone else had success by refreshing config, so definitely try that.

**Reply by wakevortex** [reply]
Ok tried deleting .autoortho, and now the program doesnt even start ! ..put it back, and it starts but then dissapears as before..
Im attaching the file to see if its somehow the problem
 Ive had to rename it as it wouldnt upload here with its old extension
 Does the .autoortho not get built on first program start if missing? 
[autoortho.txt](https://github.com/kubilus1/autoortho/files/12925006/autoortho.txt)

**Reply by wakevortex** [reply]
anything else I can try ?

**Comment by Jeff2122** [comment]
@wakevortex  Are you sure it isn't just minimizing to the bottom and not crashing? Need a log to be sure, but v0.7.0 minimizes itself.

**Reply by wakevortex** [reply]
Yes I thought that may have been the issue, but the GUI Icon disappears completely (you can see that happen in the task bar ) and the autoortho task also closes down (disappears) from windows task manager ....I posted the log in my last post above.,..that seems to give no clues ...weird. !

**Reply by HeliMan** [reply]
I tried with False and True. When I clear cache and I try to run with threading=true I get error (cannot load z_ao_eur...) during loading. With threading=False X-Plane loads scenery with no issues. I tried it a few minutes ago with ver 0.7.0 and works.

**Reply by Ty6204** [reply]
Were you able to resolve your issue? I'm experiencing the same thing.

**Comment by Jeff2122** [comment]
I have changed everything that makes sense in the .autoortho file to see if it makes a difference on the plaid tiles. Nothing matters, it is the same with every version from v0.6.0 to v07.0 including the unstable version. 
I like how v070 combines everything into one window and it minimizes out of the way. I like how it manages the folders when it shuts down. I do not like the plaid tiles. As stated before, the second time I visit the airport it is all good. I rarely visit an airport twice because my use of XP is to fly to new places each time.
I will keep it on my system for test, but I fly with v0.5.1-3 when I do a flight with FSE.

**Reply by Jeff2122** [reply]
[KSSC-KGYH.zip](https://github.com/kubilus1/autoortho/files/12910008/KSSC-KGYH.zip)
This will be my last message about this. Here I flew a short sortie in FSE with v0.7.0. The snapshot is a sample of the low res tiles picked up all along the way. I have more but the zip would be too big. I am at 5000' here. The ortho looks ok from the cockpit, but screenshots just suck. Log included. 
I never see these low res tiles with the old trusty v0.5.1-3.

**Reply by kubilus1** [reply]
It's a res vs perf issue.  The older way would have *significant* higher delay and lag, leading to lots of stuttering and occasional crash (X-Plane giving up.).  It's probably always going to be the nature of things with that tradeoff until real ortho streaming is baked into the product.  Windows also will likely always have worse perf due to the more layers and poorer OS perf.

**Reply by Jeff2122** [reply]
Every time I load a new airport, I get the plaid tile thing. But if I reload the same airport those tiles are fixed. Every time. 
What is the mechanism AO uses to fix those bad tiles? Why can't it check and fix them on the fly the first time?

**Reply by Jeff2122** [reply]
@kubilus1 I understand what you are saying. I agree that part of the problem is Windows. It doesn't bother me to wait for the scenery to catch up now and then, so long as the scenery looks good. My XP has never timed out during those episodes and they only last a minute, if that long. 
I will stick with what works as long as it works, but I will keep trying newer versions.

**Reply by kubilus1** [reply]
This article explains one such issue: https://www.pcworld.com/article/2113846/windows-11-feature-slows-down-ssds-by-up-to-45-percent-what-you-can-do-about-it.html

But it's not limited to just that.  There is a lot of extra 'protection' and layers in place from the OS.  Windows Defender (and other scanning tools that may be present).  The extra encryption/decryption (on top of what may be done at the hardware level!) and FUSE being not a built in feature.

That being said, I think there is stuff that can be done to mitigate and work around all of that.  I have a few ideas on that front.  But it does mean that generally things will just perform quite a lot better on Linux, at least for the time being.

**Reply by Jeff2122** [reply]
Ha! BDE. BitLocker. That gave me so much trouble in the past when I was a working man. I was not an IT person, but I was in their office plenty of times for them to fix that great awesome program when it refused to let us boot in. They eventually opened a ticket that has no expiration to come directly to the factory to fix it on the system immediately rather than have me tear it down to take the PC to them.
Yes Linux. I still plan to dual-boot eventually.

**Reply by rayw45** [reply]
Interesting information so thanks guys. Seems to underline that folk may be better off sticking with Windows 10 at least until these Win 11 bugs are sorted.  So far I see no benefit in upgrading to Windows 11 and see only disadvantages for most of us especially those who run XP12. So I stick with Win10 and faster performance.

**Comment by ahuet27** [comment]
Are you guys still experiencing the drop in FPS now and then when it's loading in new scenery? Is there any way to combat this?

**Reply by Jeff2122** [reply]
On occasion I do when using v0.5.1-3 but I can live with it. If you fly the same route twice, it won't happen. I also get pauses when large amounts are loading during flight. Those are scary but they always recover.
I didn't see any slowdowns or pauses when using this version v0.7.0. But the quality of the tiles is poor.

**Reply by rayw45** [reply]
Just a silly question what does the term plaid stand for.  Is it short for Plausable xx something ?? All I derive from the comments is that it is inferior quality tiles. Not sure I get them here as all looks very good with AO 0.7 other than debug - True is still essential for me.

**Reply by ahuet27** [reply]
I had those long pauses. It would pause for probably 2 to 3 minutes at a time. Make sure x plane 12 isn't running in beta mode, and instead running as the final release version.

**Reply by Jeff2122** [reply]
@rayw45 Plaid tiles are low quality tiles. Maybe I should refer to them as low quality tiles instead. Here's a good example pic I just took witvj v070. See good tiles next to not so good tiles.
![Twin Otter - 2023-10-16 07 05 27](https://github.com/kubilus1/autoortho/assets/124176597/8fefb877-9972-44ea-a0d6-121cfe9a7a90)

**Reply by rayw45** [reply]
Yes appreciate what you mean by using that term but wondered where the term plaid came from. The English dictionary says it is to do with a Scottish tartan pattern so yes I see some loose connection with the tartan pattern of tiles of different resolution ,  Anyway when I am flying say over 5000 feet the resolution being lower of course does not bother me but it does when say approaching airports or just flying low and slow to enjoy the scenery in say a Cessna 172.

**Reply by Jeff2122** [reply]
Yes, the term is used loosely for sure. I used to have some fashionable red and white plaid bell-bottom pants in the '70s so that word has stuck with me through life. I do sport some plaid pajamas now and then.
This "plaid" scenery clears up if you go up higher, but that's not how I use XP so I'm sort of stuck using the older version.

**Reply by kubilus1** [reply]
Not sure if it will help, but @Jeff2122 could you try increasing the number of 'fetch_threads' in your config.  The default is 32 but maybe try 64.  Not sure if that will help or not, and may depend on your system, but worth a try.

I am going to tweak a few things over time on this front, but there always will be somewhat of a tradeoff between quality and responsiveness.

**Reply by Jeff2122** [reply]
Thanks @kubilus1 I did give it a fair shot but no joy for me. I see others enjoying it so I will keep trying things
Today, I emptied my cache and set up for v0.7.0. Changed the threads to 64. 
AutoOrtho would not run in Python. I upgraded to v3.12 yesterday to get XPython3 installed and going. At that time I removed all other Python versions from 2.7 to 3.11. So I got v3.10 installed again, and AO ran after some Python installs and updates. I tried a few different airports and the results were the same as before, rough tiles close to my location, improve with zoom out.

**Reply by Jeff2122** [reply]
Today I saw that someone tried 128 threads and got some good results. Well, not here. It looks worse than ever, plaid tiles everywhere, and it took 10 minutes to load too. Nope, the thread count is not the answer for my system.

**Reply by Jeff2122** [reply]
Another data point - I tried threads=16 and I got no ortho at all, after about 10 minutes loading time. Very interesting. This is with v0.7.0 executable.

**Comment by ahuet27** [comment]
What's everyone's experience with this version so far? is it worth updating from version 0.6.0?

**Reply by helixcraig** [reply]
I'm experiencing very slow download rates when installing Ortho Sets. Not sure if it's related to 0.7.0 or a broader issue.

**Comment by ahuet27** [comment]
I'm missing a lot of tiles in Europe. I'm using version 6.0.1. If I set my cache to a certain number, will it stop downloading tiles if it reaches that number?

**Reply by rayw45** [reply]
Hi I have posted another thread "https://github.com/kubilus1/autoortho/discussions/416" here asking this very question and as yet not one reply ???  Please add your posts on this subject to my dedicated thread asking about the AO Cache size and what is implicated as well as asking if it can be increased with the next update  to say 1TB

**Comment by Skyabad** [comment]
![2023-10-21_18-45-46](https://github.com/kubilus1/autoortho/assets/33866286/cbceea87-ef03-4fff-a799-bd81a7d55e86)

### Hi! How to fix this the problem?  Everything is configured correctly. The same problem with another package that is installed  "z_ao_asi"
**********************************************************************************************************************************
![2023-10-21_18-39-02](https://github.com/kubilus1/autoortho/assets/33866286/cb4883d9-9d19-45bf-a568-a35ca9f2625f)
![2023-10-21_18-38-47](https://github.com/kubilus1/autoortho/assets/33866286/584ab183-0e6b-4d94-ab38-884894eb0b23)
![2023-10-21_18-42-02](https://github.com/kubilus1/autoortho/assets/33866286/02cbba0e-4e6b-400e-b61b-7f00e718c2da)
![2023-10-21_15-38-35](https://github.com/kubilus1/autoortho/assets/33866286/f160e79a-97da-4eb5-9c5e-46bd0031e0c1)

**Reply by rayw45** [reply]
I have been going on and on about this problem which the only way I can fix is in the autoortho configuration file by changing the line debug = False to debug = True but that will introduce  an overhead that will slow things down to some extent but it is a work around.  I assume you are running in Windows 10 or  maybe 11, and either with Dokan or WinFSP as either gives this same problem. 

I have spent many many many hours trying to isolate what is causing this problem posted several messages asking for any help. So far I have drawn a blank and it seems the developers here do not know the cause either.  You are though not alone as several people are reporting the same problem, certainly since version 6 (maybe 5 too but I never did update to that version).

I think this is of far more importance than low quality "plaid" tiles some people are quite rightfully reporting as the problem we have is a major one for sure.

**Reply by Skyabad** [reply]
Thanks. We'll try it. We are all essentially alone, but there are many of us. :-)

**Comment by wakevortex** [comment]
For those who had an issue (like me ) of 0.7 running then closing down after 5 seconds or so , here is how I finally fixed it..
UNinstall 0.6.1 (previous version) 
then DONT use the installer for 0.7, use the zip file download and unzip it into a directory .
Delete .autortho (settings file) 
Run the new AO  0.7 inside the folder where you unzipped it... it will ask you to recreate the paths where things are 
Then it works ! 
Trying exactly this with the 0.7 installer fails for me ...as AO doesnt even start to enable you to enter the new paths... 
The key seemed to be using the zipped version... no idea why !

**Reply by kubilus1** [reply]
Thanks for that feedback.  That is very interesting, I'll try to replicate what you are seeing.  The two versions should work the same, the installer in fact the same files as the zip, just packaged up in an installer.  Maybe once installed Windows is unhappy with the location or something, no idea.

**Reply by Jeff2122** [reply]
In my case, the installers work ok and the programs run fine. I did notice that if you install more than one version, i.e. both v0.5.1 and v0.7.0, the second will overwrite the first in the registry and the first can't be uninstalled via the Apps & Features catalog any longer. 
This could be a hint - Windows could get confused on where stuff is. I am still playing around so maybe some more hints will pop out.

**Comment by libello123** [comment]
Hi
windows10
12.07r1
europa download always hangs like this:
![Autoortho](https://github.com/kubilus1/autoortho/assets/70951562/f80ba15c-289c-4b0f-9c6b-d31c03d259d3)

**Reply by N6970R** [reply]
After a clean install, I too got all of the above error messages. I also looked at the xplane12 log file and found that autoortho had unzipped some files to a different path. Check your xplane12 log file to see where they are.
I'm not sure if, or where any folders are mounted during the unzip process, but this seems to be the source of the error.
